broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -199,35 +201,25 @@ boolean isDbOpened() {
 
   private void takeSnapshot(
       final TransientSnapshot snapshot,
-      final ActorFuture<Optional<TransientSnapshot>> transientSnapshotFuture) {
+      final ActorFuture<TransientSnapshot> transientSnapshotFuture) {
     final var snapshotTaken =
         snapshot.take(
             snapshotDir -> {
               if (db == null) {
-                LOG.error("Expected to take a snapshot, but no database was opened");
-                return false;
+                throw new StateClosedException(
```

Comment:
:x:  Not so optimal I think. What about just complete the future?

Reply:
Or will the returned future be completed with the exception, sorry might be that I missed that.

Reply:
TransientSnapshot#take will complete the future (snapshotTaken) exceptionally, and do some additional error handling. That is why it doesn't complete `transientSnapshotFuture` here.